tests/Doctrine/Tests/ORM/Proxy/ProxyFactoryTest.php

```diff
@@ -139,6 +140,29 @@ public function testFailedProxyLoadingDoesNotMarkTheProxyAsInitialized(): void
         $this->assertInstanceOf('Closure', $proxy->__getCloner(), 'The cloner wasn\'t removed');
     }
 
+    public function testExceptionWhenLoadingProxyDoesNotMarkTheProxyAsInitialized(): void
+    {
+        $persister = $this->getMockBuilder(BasicEntityPersister::class)->setMethods(['load'])->disableOriginalConstructor()->getMock();
+        $this->uowMock->setEntityPersister(ECommerceFeature::class, $persister);
+
+        $proxy = $this->proxyFactory->getProxy(ECommerceFeature::class, ['id' => 42]);
+
+        $persister
+            ->expects($this->atLeastOnce())
+            ->method('load')
+            ->willThrowException(new Exception());
+
+        try {
+            $proxy->getDescription();
+            $this->fail('An exception was expected to be raised');
+        } catch (Exception $exception) {
+        }
```

Comment:
```suggestion
        $this->expectException(Exception::class);
        $proxy->getDescription();
```

Reply:
Would make the test less precise: instead of expecting exactly `getDescription` to throw, the exception could happen anywhere, e.g. when calling `$proxy->__isInitialized()`

Reply:
`expectException()` does exactly that. The method is more precise than the outdated annotation from the older PHPUnit versions.